dcap/ql/src/quote_enclave.rs

```diff
@@ -62,10 +193,36 @@ mod test {
     use mc_sgx_dcap_ql_types::PathKind::{
         ProvisioningCertificateEnclave, QuoteProviderLibrary, QuotingEnclave,
     };
-    use std::fs;
+    use serial_test::serial;
+    use std::{ffi::c_void, fs};
     use tempfile::tempdir;
     use yare::parameterized;
 
+    /// Resets the [`PATH_INITIALIZER`] to being uninitialized.  This is *not*
+    /// thread safe so tests that make use of this need to utilize the
+    /// `#[serial]` macro.
+    fn reset_path_initializer() {
```

Comment:
I do not like this, but I would rather have unsafe test code than untested normal code.
Especially if we want to support retrying path initialization, I think we'll want tests
From task list on #145
>  Consider tests and logic to allow reattempts at initializing the paths if they fail to initialize.

Reply:
Converted to a lazy mutex